src/asset/asset.cpp

```diff
@@ -0,0 +1,157 @@
+#include <cstdio>
+
+#include "../util/log.h"
+
+#include "asset.h"
+
+namespace Asset {
+
+struct System {
+    // read directly from file
+    FileHeader  file_header;
+    AssetHeader *headers;
+    AssetData   *data;
+
+    // not read directly from file
+    u64 num_assets;
+} system = {};
```

Comment:
If you make this a global variable, we will have trouble with hot code reloading. Maybe it should be attached to the global game state.